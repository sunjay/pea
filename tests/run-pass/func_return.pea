fn foo(b) {
    let a = 3 - b;
    return a;
}

fn bar(a, b, c) {
    println!(a * b - c / a + b * 33);
    return;
}

fn main() {
    println!(foo(1));
    println!(foo(100) * 40 - 50 % 3);
    bar(3019, -29, 300);

    // Weird...but still technically valid!
    let x = return;

    // should be unreachable
    println!(b"test");
    println!(x);
}
