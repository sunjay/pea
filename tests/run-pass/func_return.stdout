2
-3882
-88508
